src/tracer/stdio_redirecting.rs

```diff
@@ -0,0 +1,96 @@
+use crate::context::Context;
+use crate::R;
+use libc::c_int;
+use nix::unistd::{close, dup2, pipe, read};
+use std::thread;
+
+type RawFd = c_int;
+
+pub struct Redirector {
+    stdout: Redirect,
+    stderr: Redirect,
+}
+
+impl Redirector {
+    pub fn new(context: &Context) -> R<Redirector> {
+        Ok(Redirector {
+            stdout: Redirect::new(context, StreamType::Stdout)?,
+            stderr: Redirect::new(context, StreamType::Stderr)?,
+        })
+    }
+
+    pub fn child(&self) -> R<()> {
```

Comment:
is there a more descriptive name for this other than "child"? similarly for "parent". i think i get what you're doing here, but the names don't give me much of a hint.

Reply:
Done. Hope these names are better now.